plugins/inputs/logfile/fileconfig.go

```diff
@@ -203,3 +251,55 @@ func (config *FileConfig) isMultilineStart(logValue string) bool {
 	}
 	return config.MultiLineStartPatternP.MatchString(logValue)
 }
+
+func (config *FileConfig) shouldFilterLog(event logs.LogEvent) bool {
+	if config.Filters == nil || len(config.Filters) < 1 {
+		return false
+	}
+	config.sampleCount += 1
+	start := time.Now()
+	for _, filter := range config.Filters {
+		if filter.ExpressionP == nil {
+			filter.ExpressionP = regexp.MustCompile(filter.Expression)
+		}
+		matched := filter.ExpressionP.MatchString(event.Message())
+		switch filter.Type {
+		case includeType:
+			if matched {
+				elapsed := time.Since(start)
+				if config.sampleCount >= sampleThreshold {
+					log.Printf("D! [inputs.logfile.%s.%s] Matched log message with expression {%s} in %dns", config.LogGroupName, config.LogStreamName, filter.Expression, elapsed.Nanoseconds())
+					config.sampleCount = 0
+				}
+				return false
+			}
+		case excludeType:
+			if matched {
+				elapsed := time.Since(start)
+				if config.sampleCount >= sampleThreshold {
+					log.Printf("D! [inputs.logfile.%s.%s] Matched log message with expression {%s} in %dns", config.LogGroupName, config.LogStreamName, filter.Expression, elapsed.Nanoseconds())
+					config.sampleCount = 0
+				}
+				profiler.Profiler.AddStats(
+					[]string{"logfile", config.LogGroupName, config.LogStreamName, "messages", "dropped"},
+					1,
+				)
+				return true
```

Comment:
Another use case to consider:
config: `[include_FOO, exclude_BAR]` will upload log lines with "FOO BAR".
config: `[exlclude_BAR, include_FOO]` will NOT upload log lines with "FOO BAR".

Did I get that right?
Is this by design?

Reply:
That was my intent, but noting Ben's comment on the implementation, may not be true going forward.